File: src/main/java/me/pcasaes/hexoids/core/application/commands/LeaveGame.java
package me.pcasaes.hexoids.core.application.commands;

import me.pcasaes.hexoids.core.domain.eventqueue.GameQueue;
import me.pcasaes.hexoids.core.domain.model.EntityId;
import me.pcasaes.hexoids.core.domain.model.Game;
import me.pcasaes.hexoids.core.domain.model.Player;

public class LeaveGame {

    private final GameQueue gameQueue;

    LeaveGame(GameQueue gameQueue) {
        this.gameQueue = gameQueue;
    }

    public void leave(EntityId userId) {
        this.gameQueue.enqueue(() -> Game.get().getPlayers().get(userId).ifPresent(Player::leave));
    }
}

File: src/main/kotlin/me/pcasaes/hexoids/core/application/commands/LeaveGame.kt
package me.pcasaes.hexoids.core.application.commands

import me.pcasaes.hexoids.core.domain.eventqueue.GameQueue
import me.pcasaes.hexoids.core.domain.model.EntityId
import me.pcasaes.hexoids.core.domain.model.Game

class LeaveGame internal constructor(private val gameQueue: GameQueue) {

    fun leave(userId: EntityId) {
        this.gameQueue.enqueue {
            Game.get().getPlayers().get(userId)?.leave()
        }
    }
}
